lib/identity/client.js

```diff
@@ -766,6 +766,51 @@ class Client extends PartialClient {
     )
     return await validateResponseAsJSON(response)
   }
+  /**
+   * removeSingleSecret
+   * @param {String} secretID
+   * @param {String} version
+   * @param {String} secretName
+   * @param {String} secretType
+   * @return {Boolean} returns true if the secret was successfully removed
+   */
+  async removeSingleSecret(secretID, version, secretName, secretType) {
+    const sharedList = await this.getSecretSharedList(secretName, secretType)
+    await Promise.all(
+      sharedList.map(async users => {
+        let identity = await this.searchIdentityByUsername(users.username)
```

Comment:
@rosasck this is a bulk endpoint right? We should be able to do this search once around 778 to get all the usernames and not have to make a search for each user the secret is shared with.

Reply:
`searchIdentitiesByUsername` is a bulk endpoint! 

@carrala  should have caught this during our debug session! we just need a list of all usernames and then call `searchIdentitiesByUsername` NOT `searchIdentityByUsername`

Reply:
in hindsight maybe too close of a name....